utils.py

```diff
@@ -0,0 +1,31 @@
+from functools import wraps
+from flask import jsonify, request
+import inspect
+from config import config
+import jwt
+
+def token_valid_check(auth_header):
+    auth_token = auth_header.split(' ')[1]
+    if auth_token == "" or auth_token == None:
+        raise Exception({'message':'Authorization token empty'})
+    decoded_token = jwt.decode(auth_token, config['ACCESS_TOKEN_SECRET'], algorithms=["HS256"])
+    return decoded_token
+
+def authenticate_user(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        auth_header = request.headers.get('Authorization')
+        payload = token_valid_check(auth_header)
+        if 'payload' in inspect.getfullargspec(func).args:
```

Comment:
not sure this is a great way to do this - can you send me some documentation about it